docs/reference/builder.md

```diff
@@ -1181,12 +1183,18 @@ For example, consider this Dockerfile:
 3 RUN echo $CONT_IMG_VER
 ```
 
-If you specify `--build-arg CONT_IMG_VER=<value>` on the command line the
-specification on line 2 does not cause a cache miss; line 3 does cause a cache
-miss. The definition on line 2 has no impact on the resulting image. The `RUN`
-on line 3 executes a command and in doing so defines a set of environment
-variables, including `CONT_IMG_VER`. At that point, the `ARG` variable may
-impact the resulting image, so a cache miss occurs.
+```
+1 FROM ubuntu
+2 ARG CONT_IMG_VER
+3 RUN echo hello
+```
+If you specify `--build-arg CONT_IMG_VER=<value>` on the command line, in both
+cases, the specification on line 2 does not cause a cache miss; line 3 does cause
+a cache miss. The definition on line 2 has no impact on the resulting image. The
+`RUN` on line 3 executes a command and in doing so defines a set of environment
+variables, including `CONT_IMG_VER` if it has a value associated with it. At
+that point, the `ARG` variable may impact the resulting image, so a cache miss
+occurs.
```

Comment:
@duglin I think 
> If a Dockerfile defines an `ARG` variable whose value is different from a previous build, then a "cache miss" occurs upon the first time it is possible to be used, not its declaration."

has already explained the reason that cache miss occurs. cache miss cache occurs when 'ARG' variable changed and the commond ('run') is possible to use it.